test/test_nn.py

```diff
@@ -9348,6 +9348,57 @@ def test_conv_noncontig_weights_and_bias(self, device):
             out2 = conv1(input_c)
             self.assertEqual(out1, out2)
 
+    @onlyCUDA
+    @dtypesIfCUDA(torch.half, torch.float, torch.double)
+    def test_grid_sample_large(self, device, dtype):
+        def issue_35202():
```

Comment:
only 2 of the subtests actually use dtype argument. The rest of the tests will be repeated 3 times doing the same thing, using hardcoded dtype value. Can you structure the tests so it does not happen

Reply:
Thanks. I will restructure the test.